elements/lisk-p2p/src/directory/new_peers.ts

```diff
@@ -0,0 +1,277 @@
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { P2PPeerInfo } from '../p2p_types';
+import { constructPeerIdFromPeerInfo, getBucket, PEER_TYPE } from '../utils';
+
+export const DEFAULT_NEW_PEER_BUCKET_COUNT = 128;
+export const DEFAULT_NEW_PEER_BUCKET_SIZE = 32;
+export const DEFAULT_ELIGIBLE_DAYS_FOR_EVICTION = 30;
+
+const MILLISECONDS_IN_ONE_DAY = 86400000; // Formula hours*minutes*seconds*milliseconds;
+
+export interface NewPeerConfig {
+	readonly eligibleDaysForEviction?: number;
+	readonly newPeerBucketCount?: number;
+	readonly newPeerBucketSize?: number;
+	readonly secret: number;
+}
+interface NewPeerInfo {
+	readonly peerInfo: P2PPeerInfo;
+	readonly dateAdded: Date;
+}
+
+export interface AddPeerOutcome {
+	readonly success: boolean;
+	readonly isEvicted: boolean;
+	readonly evictedPeer?: P2PPeerInfo;
+}
+export class NewPeers {
+	private readonly _newPeerMap: Map<number, Map<string, NewPeerInfo>>;
+	private readonly _newPeerBucketCount: number;
+	private readonly _newPeerBucketSize: number;
+	private readonly _eligibleDaysForEviction: number;
+	private readonly _secret: number;
+
+	public constructor({
+		eligibleDaysForEviction,
+		newPeerBucketSize,
+		newPeerBucketCount,
+		secret,
+	}: NewPeerConfig) {
+		this._newPeerBucketSize = newPeerBucketSize
+			? newPeerBucketSize
+			: DEFAULT_NEW_PEER_BUCKET_SIZE;
+		this._newPeerBucketCount = newPeerBucketCount
+			? newPeerBucketCount
+			: DEFAULT_NEW_PEER_BUCKET_COUNT;
+		this._eligibleDaysForEviction = eligibleDaysForEviction
+			? eligibleDaysForEviction
+			: DEFAULT_ELIGIBLE_DAYS_FOR_EVICTION;
+		this._secret = secret;
+		// Initialize the Map with all the buckets
+		this._newPeerMap = new Map();
+		[...Array(this._newPeerBucketCount).keys()]
+			.map(x => x + 1)
+			.forEach(bucketNumber => {
+				this._newPeerMap.set(bucketNumber, new Map<string, NewPeerInfo>());
+			});
+	}
+
+	public get newPeerConfig(): NewPeerConfig {
+		return {
+			newPeerBucketSize: this._newPeerBucketSize,
+			newPeerBucketCount: this._newPeerBucketCount,
+			secret: this._secret,
+		};
+	}
+
+	public get newPeerMap(): Map<number, Map<string, NewPeerInfo>> {
+		return this._newPeerMap;
+	}
+
+	public getNewPeersList(): ReadonlyArray<P2PPeerInfo> {
+		const peersListMap: P2PPeerInfo[] = [];
+
+		[...this._newPeerMap.values()].forEach(peerMap => {
+			peerMap.forEach(peer => peersListMap.push(peer.peerInfo));
+		});
+
+		return peersListMap;
+	}
+
+	public getBucketId(ipAddress: string): number {
+		return getBucket({
+			secret: this._secret,
+			peerType: PEER_TYPE.NEW_PEER,
+			targetAddress: ipAddress,
+		});
+	}
+
+	public findPeer(peerInfo: P2PPeerInfo): boolean {
```

Comment:
I think we can use `getPeers` instead of `findPeer` because almost serves the same purpose and moreover returns the peer. I will remove this function and only `getPeers`